supply-chain-security/compromises/2019/webmin-backdoor.md

```diff
@@ -0,0 +1,13 @@
+# Webmin backdoor
+
+At some time in April 2018, the Webmin development build server was exploited and a vulnerability added to one of the scripts. The vulnerable code was only present in Webmin packages offered for download via SourceForge, but not on GitHub. Because the timestamp of the modified script was set back, it did not show up in any Git diffs. The backdoor was first introduced in version 1.890 and was also included in 1.900 and 1.920.
+
+
+## Impact
+
+The attack surface have been enormous - without taking machines managed through Webmin into account. On its [GitHub page](https://github.com/webmin/webmin), the Webmin team claims their application has "over 1,000,000 installations worldwide." As of 19.08.2019 a [Shodan](https://www.shodan.io/) search query returns over 215,000 public Webmin instances, which can be attacked without needing to compromise internal networks or to bypass firewalls to reach a Webmin installation.
+
+
+## Type of compromise
+
+It appears the attackers compromised the publishing infrastructure. The Webmin team didn't specify if the "compromised build infrastructure" was referring to a compromised developer machine where the code was created, or to a compromised SourceForge account, which the hacker might have used to upload their own malicious Webmin version on SourceForge.
```

Comment:
They did. In the Webmin [page explaining the exploit](http://www.webmin.com/exploit.html) they state several times that it was a compromised build server, i.e.
> the Webmin development build server was exploited and a vulnerability added

You've also stated it was the build server in your summary above. 😄

Reply:
Yes, I was confused after reading so many different articles about the attack written before and after the Webmin security advisory.
I will change it.